sdks/java/extensions/sorter/src/test/java/org/apache/beam/sdk/extensions/sorter/ExternalSorterTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.extensions.sorter;
+
+import org.apache.beam.sdk.extensions.sorter.SorterTestUtils.SorterGenerator;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for Sorter. */
+@RunWith(JUnit4.class)
+public class ExternalSorterTest {
+
+  @Test
+  public void testEmpty() throws Exception {
+    SorterTestUtils.testEmpty(ExternalSorter.create(new ExternalSorter.Options()));
+  }
```

Comment:
Nope. Thinking about it a bit, not sure tests on that would be worth the cost?

For ExternalSorter both of the options are passed into Hadoop, so three solutions for testing I can think of are: 

1. Do some rough dependency injection with a separate hadoop wrapper class, then we could test this with a mock, but this just shifts the testing issue to the initializer class.
2. Verify that the arguments as they are used by hadoop. This would involve checking for temporary files under the specified directory and for memory...possibly some kind of memory inspection? Think the complexity/costs of these tests does not justify the benefit.
3. Use reflection to examine the appropriate private variables in Hadoop. This would be quite brittle. Having these tests be dependent on third party library implementation details is not great.

Let me know if you can think of a less crappy option 4 or have an argument that one of these is a good approach/worth it.

I did add tests to check for an exception on negative memory, also added a check for that in the InMemory and External sorter classes, which I had missed.